backends/dpdk/dpdkAsmOpt.cpp

```diff
@@ -264,5 +264,236 @@ bool ValidateTableKeys::preorder(const IR::DpdkAsmProgram *p) {
     return false;
 }
 
+const IR::Expression* CopyPropagationAndElimination::getIrreplaceableExpr(cstring str,
+                                                                        bool allowConst) {
+    if (dontEliminate.count(str))
+        return nullptr;
+    if (!str.startsWith("m."))
+        return nullptr;
+    auto expr = replacementMap[str];
+    const IR::Expression* prev = nullptr;
+    while (expr != nullptr
+        && (!allowConst ? expr->is<IR::Member>() : true)
+        && str != expr->toString()
+        && (!allowConst ? expr->toString().startsWith("m.") : true)
+        && dontEliminate.count(expr->toString()) == 0
+        && newUsesInfo[expr->toString()] == 0
+        && (expr->is<IR::Member>() ? haveSingleUseDef(expr->toString()) : true)) {
+        prev = expr;
+        expr = replacementMap[expr->toString()];
+    }
+    if (expr != nullptr &&  allowConst)
+        return expr;
+    else if (expr != nullptr && expr->is<IR::Member>())
+        return expr;
+    else
+        return prev;
+}
+
+void CopyPropagationAndElimination::markUseDef(const IR::DpdkBinaryStatement *b) {
+    usesInfo[b->src1->toString()]++;
+    usesInfo[b->src2->toString()]++;
+    defInfo[b->dst->toString()]++;
+}
+
+void CopyPropagationAndElimination::markUseDef(const IR::DpdkMovStatement *mv) {
+    defInfo[mv->dst->toString()]++;
+    usesInfo[mv->src->toString()]++;
+    replacementMap[mv->dst->toString()] = mv->src;
+}
+
+void CopyPropagationAndElimination::markUseDef(const IR::DpdkUnaryStatement *u) {
+    usesInfo[u->src->toString()]++;
+    defInfo[u->dst->toString()]++;
+}
+
+void CopyPropagationAndElimination::markUseDef(const IR::DpdkCastStatement *c) {
+    usesInfo[c->src->toString()]++;
+    defInfo[c->dst->toString()]++;
+    replacementMap[c->dst->toString()] = c->src;
```

Comment:
is the cast a noop?

Reply:
no, it's copy.

Reply:
I mean, it does never change the bit pattern?
That's strange, I would expect some casts to perform masking and others to perform sign-extension.
Like casting int<8> to int<16>.